v2/pkg/controller/mpi_job_controller.go

```diff
@@ -711,25 +748,31 @@ func (c *MPIJobController) getOrCreateConfigMap(mpiJob *kubeflow.MPIJob, isGPULa
 
 // getOrCreateWorkerService gets the workers' Service controlled by this MPIJob,
 // or creates one if it doesn't exist.
-func (c *MPIJobController) getOrCreateWorkersService(mpiJob *kubeflow.MPIJob) (*corev1.Service, error) {
-	svc, err := c.serviceLister.Services(mpiJob.Namespace).Get(mpiJob.Name + workerSuffix)
+func (c *MPIJobController) getOrCreateWorkersService(job *kubeflow.MPIJob) (*corev1.Service, error) {
+	return c.getOrCreateService(job, newWorkersService)
```

Comment:
Can you simplify the code by creating the service here and removing the newWorkersService() factory function? And same with the lanucher?

Reply:
I'm getting rid of the factory in the `getOrCreateService` function. But I have to keep the `newWorkersService` because it's used for tests.